app/web/src/components/UserProfile/UserProfile.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect, useRef, useReducer, ReactNode } from 'react'
+import { useAuth } from '@redwoodjs/auth'
+import { Link, navigate, routes } from '@redwoodjs/router'
+import ProjectsOfUser from 'src/components/ProjectsOfUserCell'
+import IdeHeader from 'src/components/IdeHeader/IdeHeader'
+import Svg from 'src/components/Svg/Svg'
+import {
+  fieldsConfig,
+  fieldReducer,
+  UserProfileType,
+  FieldConfigType,
+} from './userProfileConfig'
+
+function buildFieldsConfig(fieldsConfig, user) {
```

Comment:
Initializes values from `user` state into the config object. Is this a code smell?